### PR TITLE
First draft for `iban` and `account number/bank code` check.

### PR DESCRIPTION
**Pull request by igorkosta**
The idea is either validate `iban` and `account nr./bank code` separately:
- iban_check.json
- bank_account_check.json

or validate both by using one endpoint
- bank_account_check_draft.json

### CODE REVIEW

**Comment by schorsch** [comment]
Good question! So how do the (response) objects look like? Think of the resource we are handling: is the resource a validation information {valid: boolean, error: string} or a bank account: {iban:string, bic:string}

In the first case i would probably have one resource with two validation endpoints(links GET) with differing params(iban vs. account number). 
In the second case: Do the resources differ? e.g. i pass in an iban and get back the bic + bank name. In such a case i would prefer different resources.

Besides think of the endpoints method: Do we create a resource(POST) or do we simply GET back a piece of information.

I would probably go with a pure validation resoruce, which could be extended by new methods to check other data(with different params). following a quick scratch

```javascript

{
  "$schema"     : "http://json-schema.org/draft-04/schema#",
  "type"        : "object",
  "title"       : "bank_check",
  "description" : "Check whether your account data or IBAN is valid",  
  "properties"  : {
    "valid" : {
      "description" : "If the passed data is valid",
      "type"  : "boolean"
    },
    "error" : {
      "description" : "Error info",
      "type"  : "string"
    }
  },
  "links" : [
    {
      "description" : "Validate bank account details",
      "rel"    : "validate_bank",
      "href"   : "validate_bank",
      "method" : "GET",
      // URL params
      "properties"  : {
          "account_number" : {
            "description" : "Account number",
            "type"  : "string"
          }
        }
    },
    {
      "description" : "Validate IBAN",
      "rel"    : "validate_iban",
      "href"   : "validate_iban",
      "method" : "GET",
      "properties"  : {
          "account_number" : {
            "description" : "Account number",
            "type"  : "string"
          }
        }
    }
  ]
}

```

**Comment by MarcGrimme** [comment]
From following this discussion I could have been a bit hasty by just rushing into @igorkosta 's office requesting this ;-) . So what do we need?

## Story 1a (positive):
As an existent Fidor customer I want to validate an IBAN and get back the corresponding Bank Name, if the IBAN is correct. 
### Note to 1a: As return codes we could also include IBAN and BIC so that the return JSON is compatible to Story 2 (but decide by yourself).

## Story 1b(negative As an existent Fidor customer I want to validate an IBAN and get back an error message if the IBAN is not correct. 

## Story 2a (positive)
As an existent Fidor customer I want to validate a "Kontonummer"/"BLZ" combination and get back the corresponding Bank Name, IBAN and BIC , if the "Kontonummer"/"BLZ" combination is correct. 

### Note 2a (see Note 1a).

## Story 2b (negative): see 1b

I hope this is a better description.

**Comment by igorkosta** [comment]
@schorsch - we can got with {{GET}}. Although I'm pretty sure {{properties}} in the {{links}} section won't be recognized by the {{json}} libraries we use as {{url params}}.

**Comment by igorkosta** [comment]
Ok, so I revised my first draft. The idea is to provide the `/validates` endpoint that can validate either `iban`s or the `account_number/bank_code` pairs:

`GET https://api.fidor.de/validates?iban= DE89370400440532013000`
`GET https://api.fidor.de/validates?account_number=6666666&bank_code= 70022200 `

**Comment by MarcGrimme** [comment]
I'd change the endpoint. For me '/validates' is to generic. Can we not have '/validates/payment' or '/validates/payment_address'? Or the other way around - don't know what's better - '/payment/validate'?

**Comment by igorkosta** [comment]
Yes, sure we have to change it - it was a first draft. I was struggling with naming and still do. 
Not really sure about including the `payment` into the `URI` since we're not validating the `payment` but just the `bank account` related part of it.
Although I don't like the `bank_check` option but it seems like the shortest form of what we're trying to `check` here.

p.s. `/validates?iban` just sounded very natural :)

**Comment by weiss100** [comment]
I would opt for the  /validate/iban:iban solution and use ? only for paramters like filter, etc.. (other API providers have structures like /validate/iban as well)

**Comment by MarcGrimme** [comment]
New information. There seems to be a (legislation)[http://www.europeanpaymentscouncil.eu/index.cfm/sepa-credit-transfer/iban-and-bic/] getting into action on 1.2.2016 that will prohibit the usage of "Kontonummer/BLZ" combination. This means I'll remove the cases 2a/b.

File: schema/v1.0/bank_check.json [hunk 1]
```diff
@@ -0,0 +1,35 @@
+{
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "type": "object",
+  "title": "bank_check",
+  "description": "Validate provided bank IBAN",
+  "properties": {
+    "iban" : {
+      "$ref" : "./base_types/base_types.json#definitions/iban"
+    },
+    "bic" : {
```
Comment (MarcGrimme) [inline schema/v1.0/bank_check.json]: Hi Igor,
one tiny bit. I think we MUST NOT return the BIC because of the legal requirements.
Besides that :boat:

**Comment by MarcGrimme** [comment]
Merging this PR. I don't think we'll get more out of this.

**Pull request merged**